addons/mendeley/models.py

```diff
@@ -267,3 +269,20 @@ class NodeSettings(BaseCitationsNodeSettings):
     def _fetch_folder_name(self):
         folder = self.api._folder_metadata(self.list_id)
         return folder.name
+
+    def get_folders(self, **kwargs):
+        if self.has_auth:
+            try:
+                folders = self.api._get_folders()
+                return [{
+                    'addon': 'mendeley',
+                    'kind': 'folder',
+                    'id': folder.json['id'],
+                    'name': folder.json['name'],
+                    'path': '/'
+
+                } for folder in folders]
+            except MendeleyApiException:
```

Comment:
Do we necessarily want this to be the fail state? If an exception is thrown here, how do we know? Maybe we can log this?

Reply:
@Johnetordoff hmm not sure, I patterned this part off of your [github work](https://github.com/CenterForOpenScience/osf.io/pull/8194/files#diff-afdde548161631514df8e9d2ae22ae7aR205) .   Do you think we should send to Sentry or should we throw specific errors here that v2 can use to show specific status codes?

Reply:
I don't have any specific ideas about this, I was mostly wondering if you had any ideas. I don't think there are any specific HTTP codes for this type of thing. I was more thinking along the lines of 'We don't necessarily need the user to know if there was an error, but it would be nice to know internally if this was failing and how.' Anyway this is fine how it is, it's very similar to how other providers work.